src/lib/terkin/driver/vedirect_sensor.py

```diff
@@ -62,36 +90,69 @@ def start(self):
         try:
 
             # MicroPython
-            if platform_info.vendor in [platform_info.MICROPYTHON.Vanilla, platform_info.MICROPYTHON.Pycom]:
-                raise NotImplementedError('VEDirect driver not implemented on MicroPython')
+            if platform_info.vendor in [
+                platform_info.MICROPYTHON.Vanilla,
+                platform_info.MICROPYTHON.Pycom,
+            ]:
+                try:
+                    from vedirect import VEDirect
+
+                    uart = int(self.device)
+                    self.driver = VEDirect(serialport=uart, timeout=self.timeout)
+                except Exception as e:
+                    log.exc(
+                        e,
+                        "Could not start VEDirect interface on device: {}".format(
+                            self.device
+                        ),
+                    )
+                    raise e
 
             # CPython
             elif platform_info.vendor == platform_info.MICROPYTHON.RaspberryPi:
-                from vedirect import Vedirect
-                self.driver = Vedirect(serialport=self.device, timeout=self.timeout)
+                from vedirect import VEDirect
 
-            else:
-                raise NotImplementedError('VEDirect driver not implemented on this platform')
+                self.driver = VEDirect(serialport=self.device, timeout=self.timeout)
 
+            else:
+                raise NotImplementedError(
+                    "VEDirect driver not implemented on this platform"
+                )
+
+            if self.threading:
+                # Todo: probably need to use a queue to pass data between threads?
+                _thread.start_new_thread(self._async_read, ())
```

Comment:
Maybe move the import here in order to satisfy all platforms where this module is not available at all?
```suggestion
                import _thread
                _thread.start_new_thread(self._async_read, ())
```